tool/install_openssl.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+if [ ${#} -lt 1 ]; then
+  echo "OpenSSL version required." 1>&2
+  exit 1
+fi
+
+OPENSSL_VERSION="${1}"
+OPENSSL_DIR="${HOME}/openssl/openssl-${OPENSSL_VERSION}"
+TMP_DIR="/tmp/openssl"
+JOBS="-j$(nproc)"
+
+if [ -d "${TMP_DIR}" ]; then
+  rm -rf "${TMP_DIR}"
+fi
+mkdir -p "${TMP_DIR}"
+curl -s https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz | \
+  tar -C "${TMP_DIR}" -xzf -
+pushd "${TMP_DIR}/openssl-${OPENSSL_VERSION}"
+if [ -d "${OPENSSL_DIR}" ]; then
+  rm -rf "${OPENSSL_DIR}"
+fi
+./Configure \
+  --prefix=${OPENSSL_DIR} \
+  enable-crypto-mdebug enable-crypto-mdebug-backtrace \
+  linux-x86_64
+make -s $JOBS
+make -s install_sw
```

Comment:
This build logic is inspired from `ruby-openssl-docker` image's `Dockerfile`.
https://github.com/ruby/openssl/blob/master/tool/ruby-openssl-docker/Dockerfile